### ISSUE TITLE
Problem when shifting a date one month back and it ends up in February 30

### ISSUE THREAD

**Issue by shamanime** [bitwalker/timex#144]
### Steps to reproduce

This is the code:
```
iex(1)> Timex.Date.today
#<Date(2016-03-30)>
iex(2)> Timex.Date.today |> Timex.Date.shift(months: 1)
#<Date(2016-04-30)>
iex(3)> Timex.Date.today |> Timex.Date.shift(months: 2)
#<Date(2016-05-30)>
iex(4)> Timex.Date.today |> Timex.Date.shift(months: -2)
#<Date(2016-01-30)>
iex(5)> Timex.Date.today |> Timex.Date.shift(months: -1)
** (FunctionClauseError) no function clause matching in Timex.DateTime.resolve_timezone_info/1
    (timex) lib/datetime/datetime.ex:848: Timex.DateTime.resolve_timezone_info({:error, :invalid_date})
    (timex) lib/datetime/datetime.ex:808: Timex.DateTime.shift_by/3
    (timex) lib/datetime/datetime.ex:710: Timex.DateTime.apply_shifts/2
    (timex) lib/date/date.ex:374: Timex.Date.shift/2
iex(6)> Timex.Date.today |> Timex.Date.shift(days: -1) |> Timex.Date.shift(months: -1)
#<Date(2016-02-29)>
```
### Description of issue

It should move one month back, respecting the last day of the previous month.

Right now, seems like it tries to convert the date to 2016-02-30 and fails because February doesn't have 30 days.

**Comment by bitwalker** [comment]
Ah damn, I handle this in the other shift code, but just changed this in the Timezone module to use it's own shift implementation to avoid infinite recursion. I'll get a fix in today.

**Comment by bitwalker** [comment]
This is now fixed and pushed in to hex in 2.1.4

**Comment by shamanime** [comment]
Like a boss! Thank you!
